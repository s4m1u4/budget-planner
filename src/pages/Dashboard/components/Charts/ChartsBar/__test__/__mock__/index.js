export { TITLE, DATA } from "./mock";
