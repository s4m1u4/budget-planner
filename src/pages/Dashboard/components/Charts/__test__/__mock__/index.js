export { CATEGORIES, RECORDS } from "./mock";
