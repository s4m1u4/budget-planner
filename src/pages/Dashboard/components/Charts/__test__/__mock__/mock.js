export const RECORDS = [
  {
    id: "9cc6c0d7-24e3-46aa-aa7c-73e86c1dd230",
    amount: "2354.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "084a3558-6b14-4e38-9d24-47e912021384",
  },
  {
    id: "bc732adb-7b27-4d76-944c-4e7013eb97fa",
    amount: "836.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "084a3558-6b14-4e38-9d24-47e912021384",
  },
  {
    id: "9410db09-8e1c-4c76-a6b3-e766d9b53ae5",
    amount: "2454.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "084a3558-6b14-4e38-9d24-47e912021384",
  },
];

export const CATEGORIES = [
  {
    id: "084a3558-6b14-4e38-9d24-47e912021384",
    title: "Work",
    description: "Work",
    isDefault: false,
  },
  {
    id: "0ce314b9-a547-4f57-9d64-e466777ba8e9",
    title: "Travel",
    description: "awdawda",
    isDefault: false,
  },
  {
    id: "ea298d16-8bdb-4c5b-8198-6b2e5ce2f9a2",
    title: "Education",
    description: "Education it's so important for you!",
    isDefault: false,
  },
];
