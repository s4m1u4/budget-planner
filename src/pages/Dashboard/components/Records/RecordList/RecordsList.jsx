import React from "react";
import { RecordsItem } from "../RecordsItem";
import { RecordsListWrapper } from "./RecordsList.styles";

export const RecordsList = ({ lastRecords }) => {
  return (
    <RecordsListWrapper>
      {lastRecords.map((record) => (
        <RecordsItem key={record.id} record={record} />
      ))}
    </RecordsListWrapper>
  );
};
