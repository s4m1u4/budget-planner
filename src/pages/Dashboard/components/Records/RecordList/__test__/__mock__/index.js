export { LAST_RECORDS } from "./mock";
