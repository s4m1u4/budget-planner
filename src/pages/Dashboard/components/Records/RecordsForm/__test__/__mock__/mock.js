export const CATEGORIES = [
  {
    id: "084a3558-6b14-4e38-9d24-47e912021384",
    title: "Work",
    description: "Work",
    isDefault: false,
  },
  {
    id: "0ce314b9-a547-4f57-9d64-e466777ba8e9",
    title: "Travel",
    description: "awdawda",
    isDefault: false,
  },
  {
    id: "ea298d16-8bdb-4c5b-8198-6b2e5ce2f9a2",
    title: "Education",
    description: "Education it's so important for you!",
    isDefault: false,
  },
];
