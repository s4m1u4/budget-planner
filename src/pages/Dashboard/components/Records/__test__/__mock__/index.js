export { CATEGORIES, LAST_RECORDS } from "./mock";
