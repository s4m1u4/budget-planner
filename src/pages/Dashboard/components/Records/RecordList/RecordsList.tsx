import React, { FC } from "react";
import { RecordsItem } from "../RecordsItem";
import { IRecord } from "types";

import { RecordsListWrapper } from "./RecordsList.styles";

interface RecordsListProps {
  lastRecords: IRecord[];
}

export const RecordsList: FC<RecordsListProps> = ({ lastRecords }) => {
  return (
    <RecordsListWrapper>
      {lastRecords.map((record) => (
        <RecordsItem key={record.id} record={record} />
      ))}
    </RecordsListWrapper>
  );
};
