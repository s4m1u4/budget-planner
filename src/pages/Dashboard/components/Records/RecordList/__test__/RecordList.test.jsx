import { RecordsList } from "../RecordsList";
import { render } from "@testing-library/react";
import { LAST_RECORDS } from "./__mock__/mock";

const renderComponent = (lastRecords) => {
  return render(<RecordsList lastRecords={lastRecords} />);
};

describe("RecordList component", () => {
  it("snapshot matching", () => {
    const { baseElement } = renderComponent(LAST_RECORDS);
    expect(baseElement).toMatchSnapshot();
  });
});
