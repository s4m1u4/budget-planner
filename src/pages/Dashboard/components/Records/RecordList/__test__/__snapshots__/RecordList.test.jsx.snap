// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordList component snapshot matching 1`] = `
<body>
  <div>
    <div
      class="css-7wb1st"
    >
      <div
        class="css-gg4vpm"
      >
        <p
          class="css-0"
        >
          income
           from 
        </p>
        <span
          class="css-81iyw9"
          color="income"
        >
          +$2,354.00
        </span>
      </div>
      <div
        class="css-gg4vpm"
      >
        <p
          class="css-0"
        >
          income
           from 
        </p>
        <span
          class="css-81iyw9"
          color="income"
        >
          +$836.00
        </span>
      </div>
      <div
        class="css-gg4vpm"
      >
        <p
          class="css-0"
        >
          income
           from 
        </p>
        <span
          class="css-81iyw9"
          color="income"
        >
          +$2,454.00
        </span>
      </div>
    </div>
  </div>
</body>
`;
